### ISSUE TITLE
(Combination) of configuration options does not work

### ISSUE THREAD

**Issue by philly-vanilly** [PSPDFKit/PSPDFKit-Cordova#24]
We are trying out visual configurations of PSPDFKit (for Ionic/Angular/Cordova) and on the custom configuration options suggested [here](https://github.com/PSPDFKit/PSPDFKit-Cordova/tree/master/docs/android#options) we are getting runtime errors upon initializing the plugin, telling that a particular value is not valid (thus needed to be commented out or changed to default value). For us it worked only after initializing with empty options or reducing the number of options to what you see here (what is not commented out):

```typescript
    const options: PSPDFKitOptions = {
        backgroundColor: "#EFEFEF", // hex-color of the page background
        // disableAnnotationList: false, // hide annotation list (default: false)
        disableAnnotationNoteHinting: false, // hide small notes next to annotations that have a text set (default: false)
        // disableBookmarkEditing: false, // hide bookmark editing (default: false)
        // disableBookmarkList: false, // hide bookmark list (default: false)
        disableCopyPaste: false, // disable annotation copy/paste (default: false)
        // disableDocumentEditor: false, // hide document editor (default: false)
        // disableOutline: false, // hide the outline menu (default: false)
        // disablePrinting: false, // hide option to print (default: false)
        disableUndoRedo: false, // disable undo/redo system (default: false)
        // hidePageLabels: false, // hide page labels (if available in PDF) in page overlay and thumbnail grid (default: false)
        // // hidePageNumberOverlay: false, // hide the overlay showing the current page (default: false)
        // // hideSettingsMenu: false, // hide the settings menu (default: false)
        // // thumbnailBarMode: PSPDFKit.ThumbnailBarMode.THUMBNAIL_BAR_MODE_DEFAULT, // show static thumbnail bar. Also valid: THUMBNAIL_BAR_MODE_DEFAULT, THUMBNAIL_BAR_MODE_SCROLLABLE
        // // hideThumbnailGrid: false, // hide the thumbnail grid menu (default: false)
        pageFitMode: PSPDFKit.PageFitMode.FIT_TO_WIDTH, // also valid: PSPDFKit.PageFitMode.FIT_TO_SCREEN
        // scrollDirection: PSPDFKit.PageScrollDirection.VERTICAL, // also valid: PSPDFKit.PageScrollDirection.HORIZONTAL
        // scrollMode: PSPDFKit.ScrollMode.CONTINUOUS, // also valid: PSPDFKit.ScrollMode.PER_PAGE
        // // Configures which share actions should be visible in the user interface. (default: all enabled)
        shareFeatures: [
            /** Document sharing inside the activity. */
            PSPDFKit.ShareFeatures.DOCUMENT_SHARING,
            /** Sharing of embedded files (on file annotations). */
            PSPDFKit.ShareFeatures.EMBEDDED_FILE_SHARING,
            /** Sharing of text from selected free text annotations. */
            PSPDFKit.ShareFeatures.FREE_TEXT_ANNOTATION_SHARING,
            /** Sharing of selected image annotations. */
            PSPDFKit.ShareFeatures.IMAGE_SHARING,
            /** Sharing of text from selected note annotations. */
            PSPDFKit.ShareFeatures.NOTE_ANNOTATION_SHARING,
            /** Sharing of text from annotation contents or comments. */
            PSPDFKit.ShareFeatures.NOTE_EDITOR_CONTENT_SHARING,
            /** Sharing of selected text. */
            PSPDFKit.ShareFeatures.TEXT_SELECTION_SHARING
        ],
        invertColors: false, // invert rendered colors (default: false)
        toGrayscale: false, // render document in grayscale only (default: false)
        // startZoomScale: 2.0, // initial zoom value (default: 1.0)
        // maxZoomScale: 10.0, // maximum zoom factor when zooming into a page (default: 15.0)
        zoomOutBounce: true, // "bounce" animation when pinch-zooming out (default: true)
        // page: 2, // initial page number (default: 0, i.e. the first page)
        useImmersiveMode: false, // activate Android's immersive app mode (default: false)
        // disableSearch: false, // completely deactivate document search (default: false)
        // searchType: PSPDFKit.SearchType.SEARCH_MODULAR, // also valid: PSPDFKit.SearchType.SEARCH_INLINE
        // autosaveEnabled: true, // automatically save document changes on exit (default: true)
        // annotationEditing: {
        //     enabled: true, // activate annotation editing (default: true)
        //     creatorName: "John Doe" // author name written into new annotations (default: null)
        // }
    };
        PSPDFKit.present(path, options, () => {}, () => {});
```
Do the options work only in certain combinations or are they deprecated and the docs need to be updated?

We are using `"pspdfkit-cordova": "1.0.9"` with ionic 4 and cordova 9

**Comment by philly-vanilly** [comment]
This is on Android. For example, if I uncomment hidePageLabels in the code above, I get:

`E/PluginManager: Uncaught exception from plugin
    com.pspdfkit.cordova.PSPDFKitPluginException: Error while parsing option 'hidePageLabels'
        at com.pspdfkit.cordova.action.document.ShowDocumentAction.parseOptionsToConfiguration(ShowDocumentAction.java:187)
        at com.pspdfkit.cordova.action.document.ShowDocumentAction.execAction(ShowDocumentAction.java:197)
        at com.pspdfkit.cordova.action.BasicAction.exec(BasicAction.java:41)
        at com.pspdfkit.cordova.action.ActionManager.executeAction(ActionManager.java:29)
        at com.pspdfkit.cordova.PSPDFKitPlugin.execute(PSPDFKitPlugin.java:128)
        at org.apache.cordova.CordovaPlugin.execute(CordovaPlugin.java:98)
        at org.apache.cordova.PluginManager.exec(PluginManager.java:132)
        at org.apache.cordova.CordovaBridge.jsExec(CordovaBridge.java:59)
        at org.apache.cordova.engine.SystemExposedJsApi.exec(SystemExposedJsApi.java:41)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:336)
        at android.os.Looper.loop(Looper.java:174)
        at android.os.HandlerThread.run(HandlerThread.java:67)
     Caused by: java.lang.IllegalArgumentException: Invalid plugin option 'hidePageLabels'
        at com.pspdfkit.cordova.action.document.ShowDocumentAction.parseOptionsToConfiguration(ShowDocumentAction.java:183)
        at com.pspdfkit.cordova.action.document.ShowDocumentAction.execAction(ShowDocumentAction.java:197) 
        at com.pspdfkit.cordova.action.BasicAction.exec(BasicAction.java:41) 
        at com.pspdfkit.cordova.action.ActionManager.executeAction(ActionManager.java:29) 
        at com.pspdfkit.cordova.PSPDFKitPlugin.execute(PSPDFKitPlugin.java:128) 
        at org.apache.cordova.CordovaPlugin.execute(CordovaPlugin.java:98) 
        at org.apache.cordova.PluginManager.exec(PluginManager.java:132) 
        at org.apache.cordova.CordovaBridge.jsExec(CordovaBridge.java:59) 
        at org.apache.cordova.engine.SystemExposedJsApi.exec(SystemExposedJsApi.java:41) 
        at android.os.MessageQueue.nativePollOnce(Native Method) 
        at android.os.MessageQueue.next(MessageQueue.java:336) 
        at android.os.Looper.loop(Looper.java:174) 
        at android.os.HandlerThread.run(HandlerThread.java:67)
D/SystemWebChromeClient: http://localhost/plugins/pspdfkit-cordova/www/PSPDFKit.js: Line 1215 : Error while parsing option 'hidePageLabels'
I/chromium: [INFO:CONSOLE(1215)] "Error while parsing option 'hidePageLabels'", source: http://localhost/plugins/pspdfkit-cordova/www/PSPDFKit.js (1215)
`

**Comment by davidschreiber** [comment]
Hi @philly-vanilly 👋 Thanks for reporting this. I had a look at your issue, and it is indeed a problem with our internal Android configuration mapping. I went ahead and created PR #25 that should fix the issue. It is currently in review, and we will publish a new release once it is merged. 

You can already test the fix before the new release is out by loading the plugin directly from the branch, like so:

```bash
cordova plugin remove pspdfkit-cordova
cordova plugin add https://github.com/PSPDFKit/PSPDFKit-Cordova.git#davidschreiber/fix-android-configuration-adapter
```

Hope this helps!

**Comment by davidschreiber** [comment]
We just merged the fix and published a new version `1.0.11` to npm. Thanks, once more for the report!
